### PR TITLE
Add maxCollateralInputs to ProtocolParameters

### PR DESCRIPTION
**Pull request by sevanspowell**
# Issue Number

ADP-958

# Overview

- Included the maxCollateralInputs protocol parameter in the ProtocolParameters data type.

# Comments

- No tests exist for this structure, I believe this structure is tested implicitly via the `ApiNetworkParameters` structure. But since modifying that structure would encroach on ADP-1061, I'm going to leave this without tests for now. If anyone can think of a test they'd like to see, don't hesitate to suggest one!

### CODE REVIEW

File: lib/shelley/src/Cardano/Wallet/Shelley/Compatibility.hs [hunk 1]
```diff
@@ -1072,6 +1073,18 @@ unsafeToWord64 n
   where
     crash x = error ("Internal error: Word64 " ++ x)
 
+-- | Convert an int to 'Word16'.
+--
+-- If this conversion would under/overflow, there is not much we can do except
+-- to hastily exit.
+unsafeToWord16 :: Integral n => n -> Word16
```
Comment (rvl) [inline lib/shelley/src/Cardano/Wallet/Shelley/Compatibility.hs]: Perhaps you could combine this with `unsafeToWord64` and make
```suggestion
unsafeToWord :: (Integral n, Integral w) => n -> w
```

**Review by rvl — APPROVED** [review]
Yep all good - hopefully some test case needs updating for the new field.

**Review by rvl — APPROVED** [review]
Great, thanks. I added a test case for `unsafeIntToWord`, just to be sure, and took the opportunity to improve the way we crash the app.

**Comment by sevanspowell** [comment]
bors retry

**Comment by sevanspowell** [comment]
bors retry

**Comment by rvl** [comment]
Oops, sorry I pushed a minor fix to `.github/ISSUE_TEMPLATES/config.yml` directly to master and tripped up bors. Will just merge now this since it's already passed bors.

**Pull request merged**